dsgrid/cli/dsgrid_admin.py

```diff
@@ -220,10 +222,28 @@ def remove_project(registry_manager, project_id):
 
 @click.command(name="remove")
 @click.argument("dataset-id")
+@click.argument("version")
 @click.pass_obj
-def remove_dataset(registry_manager, dataset_id):
+def remove_dataset(registry_manager: RegistryManager, dataset_id: str, version: str):
     """Remove a dataset from the dsgrid repository."""
-    registry_manager.dataset_manager.remove(dataset_id)
+    registry_manager.dataset_manager.remove(dataset_id, version=version)
+    project_mgr = registry_manager.project_manager
+    for project_id in project_mgr.list_ids():
+        config = project_mgr.get_by_id(project_id)
+        for dataset in config.iter_datasets():
+            if (
+                dataset.dataset_id == dataset_id
+                and dataset.version == version
+                and dataset.status == DatasetRegistryStatus.REGISTERED
+            ):
+                dataset.status = DatasetRegistryStatus.UNREGISTERED
+                msg = (
+                    f"Set dataset {dataset_id} status to unregistered in project {project_id} "
+                    "after removal."
+                )
+                project_mgr.update(
+                    config, VersionUpdateType.MINOR, msg, submitter=getpass.getuser()
+                )
```

Comment:
I'm surprised that this is it. Do we need to unlist the data filepath or anything else like that? I was kind of expecting to see some more undo/remove code in project_registry_manager.py

Reply:
On line 228 we call into the dataset registry manager and delete the data.